src/routes/activitypub/inbox.js

```diff
@@ -66,7 +66,7 @@ async function handleUnfollow(req, res) {
 
   const myURL = new URL(req.body.actor);
   const targetDomain = myURL.hostname;
-  const name = req.body.object.replace(`https://${domain}/u/`, '');
+  const name = req.body.object.object.replace(`https://${domain}/u/`, '');
```

Comment:
Assuming Mastodon is sending the right payload according to the ActivityPub spec, this fixes it for me.